dev/tasks/build/builder.js

```diff
@@ -0,0 +1,238 @@
+		var that = this;
+		var stepCounter = 0;
+
+		// Before starting, run the initial checkups.
+		if ( !this.checkUp() ) {
+			console.log( 'Build operation aborted.' );
+			callback( false );
+
+			return;
+		}
+
+		console.log( 'Building CKEditor into the "' + this.target + '" directory:')
+
+		runNext();
+
+		function runNext() {
+			var next = that.taskList.shift();
+
+			if ( next ) {
+				stepCounter++;
+				console.log( stepCounter + '. ' + next[ 1 ] );
+				that.tasks[ next[ 0 ] ].call( that, runNext );
+			} else {
+				if ( callback ) {
+					callback();
+				}
+			}
+		}
+	},
+
+	checkUp: function() {
+		var fs = require( 'fs' );
+
+		// Stop if the tmp folder already exists.
+		if ( fs.existsSync( this.tmp ) ) {
+			console.log( 'The "' + this.tmp + '" directory already exists. Delete it and try again.' );
+
+			return false;
+		}
+
+		return true;
+	},
+
+	/**
+	 * Holds individual methods for each task executed by the builder.
+	 *
+	 * All methods here MUST be called in the builder context by using
+	 * `builder.tasks.someMethod.call( builder, callback )`.
+	 */
+	tasks: {
+		/**
+		 * Deletes the `target` and `tmp` directories.
+		 *
+		 * @param {Function} callback Function to be called when the task is done.
+		 * @returns {Object} The callback returned value.
+		 */
+		cleanUp: function( callback ) {
+			var del = require( 'del' );
+			del.sync( this.target );
+			del.sync( this.tmp );
+
+			return callback();
+		},
+
+		/**
+		 * Copy the local source code of CKEditor and its dependencies to the `tmp` directory for processing.
+		 *
+		 * @param {Function} callback Function to be called when the task is done.
+		 * @returns {Object} The callback returned value.
+		 */
+		copyToTmp: function( callback ) {
+			var ncp = require( 'ncp' ).ncp;
+			var path = require( 'path' );
+			var fs = require( 'fs' );
+			var tmp = this.tmp;
+
+			var deps = require( '../../../package.json' ).dependencies;
+
+			var toCopy = Object.keys( deps ).filter( function( name ) {
+				return name.indexOf( 'ckeditor5-' ) === 0;
+			} );
+
+			fs.mkdirSync( tmp );
+
+			function copy() {
+				var module = toCopy.shift();
+
+				if ( !module ) {
+					return callback();
+				}
+
+				var dest = path.join( tmp + '/', module );
+
+				if ( !fs.existsSync( dest ) ) {
+					fs.mkdirSync( dest );
+				}
+
+				// Copy the "src" directory only.
+				ncp( path.join( 'node_modules', module, 'src' ), path.join( dest, 'src' ), {
+					dereference: true
+				}, function( err ) {
+					if ( err ) {
+						throw err;
+					}
+
+					copy();
+				} );
+			}
+
+			copy();
+		},
+
+		/**
+		 * Removes the `CKEDITOR` namespace from AMD calls in the `tmp` copy of the source code.
+		 *
+		 * @param {Function} callback Function to be called when the task is done.
+		 * @returns {Object} The callback returned value.
+		 */
+		removeAmdNamespace: function( callback ) {
+			var replace = require( 'replace' );
+
+			replace( {
+				regex: /^\s*CKEDITOR\.(define|require)/mg,
```

Comment:
Do we have to do this? Can't the minified code be using `define` and `require` through the `CKEDITOR` namespace?